File: backend/Vasthra/src/main/java/com/vasthra/service/ProductService.java
package com.vasthra.service;

import java.util.List;

import com.vasthra.dto.ProductDTO;

public interface ProductService {

    ProductDTO createProduct(ProductDTO productDTO);

    ProductDTO updateProduct(Integer id, ProductDTO productDTO);

    ProductDTO getProductById(Integer id);

    List<ProductDTO> getAllProducts();

    List<ProductDTO> getProductsByCategory(Integer categoryId);

    void deleteProduct(Integer id);
}

File: frontend/VastraFrontEnd/src/services/productService.js
// src/services/productService.js
import axios from "axios";

const API_URL = "http://localhost:8081/api/products";

export const getAllProducts = async () => {
  const res = await axios.get(API_URL);
  return res.data;
};

export const getProductById = async (id) => {
  const res = await axios.get(`${API_URL}/${id}`);
  return res.data;
};

export const getProductsByCategory = async (categoryId) => {
  const res = await axios.get(`${API_URL}/category/${categoryId}`);
  return res.data;
};

export const createProduct = async (product) => {
  const res = await axios.post(API_URL, product);
  return res.data;
};

export const updateProduct = async (id, product) => {
  const res = await axios.put(`${API_URL}/${id}`, product);
  return res.data;
};

export const deleteProduct = async (id) => {
  await axios.delete(`${API_URL}/${id}`);
};
